### PR TITLE
MNGSITE-400 Introduce several fixes into the `POM Reference` document

### PR DESCRIPTION
**Pull request by EPadronU**
There are several typos to be fixed and improvement opportunities to be done.

### CODE REVIEW

File: content/apt/pom.apt [hunk 1]
```diff
@@ -139,12 +139,12 @@ POM Reference
   as the developers involved and the roles they play, the defect tracking system, the organization and
   licenses, the URL of where the project lives, the project's dependencies, and all of the other little
   pieces that come into play to give code life. It is a one-stop-shop for all things concerning the project.
-  In fact, in the Maven world, a project need not contain any code at all, merely a <<<pom.xml>>>.
+  In fact, in the Maven world, a project needs not contain any code at all, merely a <<<pom.xml>>>.
```
Comment (elharo) [inline content/apt/pom.apt]: "need" is correct here but "does not need to" would be less stilted

https://www.quora.com/What-is-the-way-to-correct-this-sentence-%E2%80%9CHe-needs-not-call-me%E2%80%9D

File: content/apt/pom.apt [hunk 2]
```diff
@@ -139,12 +139,12 @@ POM Reference
...
 
 * {Quick Overview}
 
   This is a listing of the elements directly under the POM's project element. Notice that <<<modelVersion>>>
-  contains 4.0.0. That is currently the only supported POM version for both Maven 2 & 3, and is always required.
+  contains 4.0.0. That is currently the only supported POM version for Maven 3, and is always required.
```
Comment (elharo) [inline content/apt/pom.apt]: delete "for Maven 3"

File: content/apt/pom.apt [hunk 3]
```diff
@@ -218,7 +218,7 @@ POM Reference
 
 * {Maven Coordinates}
 
-  The POM defined above is the minimum that Maven allows. <<<groupId:artifactId:version>>> are all
+  The POM defined above is the bare-minimum that Maven allows. <<<groupId:artifactId:version>>> are all
```
Comment (elharo) [inline content/apt/pom.apt]: no hyphen

File: content/apt/pom.apt [hunk 4]
```diff
@@ -300,8 +300,8 @@ POM Reference
 
   The cornerstone of the POM is its {{{/ref/current/maven-model/maven.html#class_dependency}dependency}} list.
   Most projects depend on others to build
-  and run correctly. If all Maven does for you is manage this list, you have gained a lot.
-  Maven downloads and links the dependencies on compilation and other goals that require them.
+  and run correctly. If all Maven does for you is to manage this list, you have already gained a lot.
```
Comment (elharo) [inline content/apt/pom.apt]: no "to"
already isn't wrong but isn't needed

File: content/apt/pom.apt (hunk 1 above)
Comment (EPadronU) [reply content/apt/pom.apt]: Got it. I agree with `does not need to`. Do you want me to change all of them? There are a couple of `need not`.

**Pull request merged**